modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/LogicalTestRunner.java

```diff
@@ -0,0 +1,285 @@
+        restartCluster = env.restart();
+        timeout = env.timeout();
+    }
+
+    /** {@inheritDoc} */
+    @Override public Description getDescription() {
+        return Description.createSuiteDescription(testCls.getName(), "scripts");
+    }
+
+    /** {@inheritDoc} */
+    @Override public void run(RunNotifier notifier) {
+        try {
+            if (script == null) {
+                Files.walk(scriptsRoot).sorted().forEach((p) -> {
+                    if (p.equals(scriptsRoot))
+                        return;
+
+                    if (Files.isDirectory(p)) {
+                        if (!F.isEmpty(Ignition.allGrids()) && restartCluster) {
+                            log.info(">>> Restart cluster");
+
+                            Ignition.stopAll(false);
+                        }
+
+                        if (F.isEmpty(Ignition.allGrids()))
+                            startCluster();
+
+                        return;
+                    }
+
+                    String dirName = p.getParent().toString().substring(scriptsRoot.toString().length() + 1);
+                    String fileName = p.getFileName().toString();
+
+                    if (!fileName.endsWith("test") && !fileName.endsWith("test_slow"))
+                        return;
+
+                    Ignite ign = F.first(Ignition.allGrids());
+
+                    for (String cacheName : ign.cacheNames())
+                        ign.destroyCache(cacheName);
+
+                    Description desc = Description.createTestDescription(dirName, fileName);
+
+                    notifier.fireTestStarted(desc);
+
+                    try {
+                        QueryEngine engine = Commons.lookupComponent(
+                            ((IgniteEx)ign).context(),
+                            QueryEngine.class
+                        );
+
+                        ScriptTestRunner scriptTestRunner = new ScriptTestRunner(p, engine, log);
+
+                        log.info(">>> Start: " + dirName + "/" + fileName);
+
+                        runScript(scriptTestRunner);
+                    }
+                    catch (Throwable e) {
+                        notifier.fireTestFailure(new Failure(desc, e));
+                    }
+                    finally {
+                        log.info(">>> Finish: " + dirName + "/" + fileName);
+                        notifier.fireTestFinished(desc);
+                    }
+                });
+            }
+            else {
+                startCluster();
+
+                runTest(script, notifier);
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        finally {
+            Ignition.stopAll(false);
+        }
+    }
+
+    /** */
+    private void runTest(Path test, RunNotifier notifier) {
+        String dirName = test.getParent().toString().substring(scriptsRoot.toString().length() + 1);
```

Comment:
```suggestion
        String dirName = test.subpath(scriptsRoot.getNameCount(), p.getNameCount() - 1).toString();
```

But please be aware that here should be additional handling for the case when test.parent == scriptsRoot